llama_stack/distribution/build_venv.sh

```diff
@@ -49,61 +47,117 @@ ENVNAME=""
 SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
 source "$SCRIPT_DIR/common.sh"
 
+# check if a command is present
+is_present() {
+  command -v "$1" &>/dev/null
+}
+
+# pre-run checks to make sure we can proceed with the installation
+pre_run_checks() {
+  local env_name="$1"
+
+  # checking if uv is installed
+  if ! is_present uv; then
+    echo "uv is not installed, trying to install it."
+    # try to install with pip
+    if ! is_present pip; then
+      echo "pip is not installed, cannot automatically install 'uv'."
+      echo "Follow this link to install it:"
+      echo "https://docs.astral.sh/uv/getting-started/installation/"
+      exit 1
+    else
+      pip install uv
+    fi
+  fi
+
+  # checking if an environment with the same name already exists
+  if [ -d "$env_name" ]; then
+    echo "Environment '$env_name' already exists, please remove it."
+    echo "Remove this directory '$env_name'."
+    exit 1
+  fi
+
+  # checking if we are in a venv and if so, deactivate it
+  if [ -n "$VIRTUAL_ENV" ]; then
+    echo "Deactivating existing virtual environment."
+    if [ -f "$VIRTUAL_ENV/bin/activate" ]; then
+      # re-hydrate the env first to make sure we can deactivate it
+      # shellcheck source=/dev/null
+      source "$VIRTUAL_ENV/bin/activate"
+      deactivate
+    else
+      echo "'deactivate' function not found."
+    fi
+  fi
+}
+
 run() {
   local env_name="$1"
   local pip_dependencies="$2"
   local special_pip_deps="$3"
 
-  pip install uv
+  echo "Creating new virtual environment $env_name"
+  uv venv "$env_name"
+  # shellcheck source=/dev/null
+  source "$env_name/bin/activate"
   if [ -n "$TEST_PYPI_VERSION" ]; then
     # these packages are damaged in test-pypi, so install them first
     uv pip install fastapi libcst
+    # shellcheck disable=SC2086
```

Comment:
can you explain necessity of this shellcheck disablement?

Reply:
It’s necessary to prevent ShellCheck from raising a warning, as this case falls under an exception. Here, $part contains words (including spaces), and if we double-quote it, it will be treated as a single command, causing the shell to fail. Therefore, word splitting is intentional in this context.

Also, the comment below the disablement explain it too :)